### PR TITLE
Fix tag release workflow

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

The current release workflow was failing with:

```
Run actions/github-script@v7
SyntaxError: Unexpected number
    at new AsyncFunction (<anonymous>)
Error: Unhandled error: SyntaxError: Unexpected number
    at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:35424:16)
    at main (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:35522:26)
    at /home/runner/work/_actions/actions/github-script/v7/dist/index.js:35497:1
    at /home/runner/work/_actions/actions/github-script/v7/dist/index.js:35553:3
    at Object.<anonymous> (/home/runner/work/_actions/actions/github-script/v7/dist/index.js:35556:12)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
```

Putting the version variable inside `"` to make it a string solves this issue.

We also check if the version variable in `version.py` increased. If that is the case, then we create a new release. Otherwise, nothing happens.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.md`

## How to test

As far as I am aware, the only way to test would be by using your own repository. Just do the following:

- Have a `share/version.py` file there.
- Copy the release workflow to that repository.

Then make the commits to check everything is working as expected. You need to update the `version.py` file to trigger the workflow, unless you also change the trigger or make the workflow run manually.

It is not possible to test it here, because we would have to update the `version.py` file of ESF, which does not make sense in this case.

### CODE REVIEW

**Review by gizas — APPROVED** [review]

**Review by axw — DISMISSED** [review]
LGTM. I'm wondering if we should make the tagging a `workflow_dispatch` workflow that one needs to explicitly run. Then we could update the version ahead of the release, and it would be present in pre-release builds and smoke tests. Doesn't need to be addressed in this PR.

**Comment by constanca-m** [comment]
I updated the workflow to only release a new tag if the new version in `version.py` is bigger than the latest released version. This way, if there was no change to the variable version in `version.py`, we also do not release a new tag, or if the version decreased somehow, we do not release it as well. @axw @gizas, is it better this way?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -20,22 +20,55 @@ jobs:
...
+          # We will list all tags sorted by the version after the prefix lambda-v.
+          # We retrieve only the first line - that is, the most recent version.
+          # After that, we remove the prefix to only get the version number.
+          old_version=$(git tag --list --sort=-version:refname "lambda-v*" | awk 'NR==1{print $1}' | awk -F"lambda-v" '{ print $NF }')
+
+          # We now need to compare the current version inside version.py.
+          IFS='.' read -a new_numbers <<< ${{ env.VERSION }}
+          IFS='.' read -a old_numbers <<< $old_version
+
+          create_tag=false # only create tag if version increased
```
Comment (gizas) [inline .github/workflows/release.yml:46]:
```suggestion
          CREATE_TAG=false # only create tag if version increased
```

(I think you can use the final variable)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -20,22 +20,55 @@ jobs:
...
+          for i in 0 1 2
+          do
+            if [[ ${new_numbers[i]} > ${old_numbers[i]} ]]
+            then
+              create_tag=true
```
Comment (gizas) [inline .github/workflows/release.yml:51]:
```suggestion
              CREATE_TAG==true
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -20,22 +20,55 @@ jobs:
...
+              break
+            elif [[ ${new_numbers[i]} < ${old_numbers[i]} ]]
+            then
+              break
+            fi
+          done
+
+          echo "CREATE_TAG=${create_tag}" >> $GITHUB_ENV
```
Comment (gizas) [inline .github/workflows/release.yml:59]:
```suggestion
          echo "CREATE_TAG=${CREATE_TAG}" >> $GITHUB_ENV
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -20,22 +20,55 @@ jobs:
...
+          old_version=$(git tag --list --sort=-version:refname "lambda-v*" | awk 'NR==1{print $1}' | awk -F"lambda-v" '{ print $NF }')
+
+          # We now need to compare the current version inside version.py.
+          IFS='.' read -a new_numbers <<< ${{ env.VERSION }}
+          IFS='.' read -a old_numbers <<< $old_version
+
+          create_tag=false # only create tag if version increased
+          for i in 0 1 2
+          do
+            if [[ ${new_numbers[i]} > ${old_numbers[i]} ]]
```
Comment (gizas) [inline .github/workflows/release.yml:49]: Does this compare only the minor version?

For example if version changes from 1.2.1 to 2.1.0 will it work?
(sorry for being lazy and not testing )

Comment (constanca-m) [reply .github/workflows/release.yml:49]: Yes, it iterates over every number between the `.`

**Review by gizas — APPROVED** [review]

**Review by axw — APPROVED** [review]
Looks good!

**Pull request merged**